UnityProject/Assets/Scripts/Core/Lifecycle/PlayerSpawn.cs

```diff
@@ -257,13 +270,27 @@ private static void ApplyNewSpawnRoleToBody( GameObject body, Occupation request
 			playerSprites.OnCharacterSettingsChange(toUseCharacterSettings);
 		}
 
-		body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
+
 		//determine where to spawn them
 		physics.AppearAtWorldPositionServer(GetSpawnPointForOccupation(requestedOccupation));
 
 		switch (spawnType)
 		{
 			case SpawnType.NewSpawn:
+				body.GetComponent<DynamicItemStorage>()?.SetUpOccupation(requestedOccupation);
```

Comment:
```suggestion
				body.GetComponent<DynamicItemStorage>().OrNull()?.SetUpOccupation(requestedOccupation);
```

Reply:
this switch looks rather busy, maybe move the body of the cases to private methods?